libraries/AP_RCProtocol/AP_RCProtocol_config.h

```diff
@@ -79,3 +79,6 @@
 #define AP_RCPROTOCOL_MAVLINK_RADIO_ENABLED AP_RCPROTOCOL_BACKEND_DEFAULT_ENABLED && BOARD_FLASH_SIZE > 1024 && HAL_GCS_ENABLED
 #endif
 
+#ifndef AP_OSD_CRSF_EXTENSIONS_ENABLED
```

Comment:
Shouldn't this be in `AP_OSD/AP_OSD_config.h` ?

Reply:
If you *really* want it in `AP_RCProtoco_config.h` then you can rename it to `AP_RCPROTOCOL_CRSG_OSD_EXTENSIONS_ENABLED`.

But I do believe it's better to be in `AP_OSD_config.h`; you can include `AP_RCProtocol_config.h` to get `AP_RCPROTOCOL_CRSF_ENABLED`.

Additionally, while at the moment it might depend on the AP_RCProtocol library, this is actually probably not the correct structure.  You can see this in the CRSF "stand-alone" mode where we only process telemetry data, not consume RC, from the CRSF serial stream.  Presumably these OSD extensions would be one use case for that.  Our current structure has the RCProtocol library process the stream, a better structure *may* be to have a separate library that parses the serial stream and pokes data into AP_RCProtocol, or into the OSD library or whatever as appropriate.  This is a long-term project, but I think shows that your assertion that this is primarily an "RC" feature is incorrect.

Reply:
Indeed, under that notion of separating the telemetry from the actual RC commands stream my assertion is incorrect. However, I made that assertion based on how things are currently implemented. But yes, you totally have a point that this really depends on the telemetry stream, being independent from the RC commands.

I will look into making this change as you suggested, then I would include `AP_OSD_config.h` into the `AP_RCProtocol_CRSF` files instead of the other way around. And remove the define from `AP_RCProtocol_config.h`.